### PR TITLE
Start resources on demand

### PR DESCRIPTION
**Pull request by praiskup**
Missing stuff
- [x] test is missing
- [x] documentation is missing

### CODE REVIEW

File: resallocserver/manager.py [hunk 1]
```diff
@@ -111,6 +111,40 @@ def run_command(pool_id, res_id, res_name, id_in_pool, command, ltype='alloc',
...
+            })
+        elif isinstance(tag, dict):
+            new_tags.append({
+                "name": tag["name"],
+                "priority": tag.get('priority', 0)
+            })
+        else:
+            assert False
+    tags.clear()
+    tags.extend(new_tags)
```
Comment (FrostyX) [inline resallocserver/manager.py:137]: This function deserves some tests. It is very testing-friendly.

**Review by FrostyX — COMMENTED** [review]
There is nothing that I have a problem with, so +1 from me.

But I must be missing something because the PR is much more complicated than I would expect.


As "resources on demand" I would expect something like `--on-demand` parameter or `--tag on-demand` for the `resalloc ticket` command. Which would basically just set high priority and ignore all limits to spawn the resource ASAP.


The linked Copr issue says "Provide a few high-performance builders (**for demanding software** like chromium)" which is IMHO a different concept than on-demand resources. But in this case, I would expect only a new pool configuration that would provide beefy machines.

This PR seems to do more things. Maybe let's discuss on some standup.

File: resallocserver/priority_queue.py [hunk 2]
```diff
@@ -2,6 +2,11 @@
 Python priority queue implementation.
 """
 
+# TODO: rename "Task" to "Item" if we want to make this a library, we can
+# place any kind of resource into the Queue (Pools, Tickets, etc.).
+# TODO: enforce the PriorityQueueTask use.  Using the default __repr__ for
```
[the remark this answers is not in the document]
Comment (praiskup) [reply resallocserver/priority_queue.py:7]: I'd prefer keeping these TODO entries.

**Review by FrostyX — APPROVED** [review]
Oh, sorry, I started reviewing but forgot to finish it.

> But I must be missing something because the PR is much more complicated than I would expect.

We discussed it last week on 1on1 call, so it makes more sense now.

**Pull request merged**